R/validation.R

```diff
@@ -0,0 +1,43 @@
+validate_date <- function(date) {
+  date_ld <- lubridate::dmy(date, quiet = TRUE)
+  date_template <- lubridate::stamp("1 January 2020", orders = "dmy", quiet = TRUE)
+  if (is.na(date_ld)) {
+    valid <- FALSE
+    validation_message <- "not in a valid date format."
+  } else if (date_ld > Sys.time()) {
+    valid <- FALSE
+    validation_message <- "is in the future."
+  } else {
+    valid <- TRUE
+    validation_message <- "is a valid date."
+  }
+  if (!valid) {
+    stop(paste(date, validation_message))
+  }
+  return(date_template(date_ld))
+}
+
+is_valid_repo_url <- function(url) {
```

Comment:
Worth adding a little extra description / reasoning on this function existing? I know it's not exported but will be helpful for future maintainers

Reply:
Done